src/domain/sales-channels/pages/details.tsx

```diff
@@ -0,0 +1,458 @@
+          <span className="text-small text-grey-50">
+            {salesChannel.description}
+          </span>
+        </div>
+      </div>
+      {isDisabled && (
+        <div className="flex flex-col justify-center">
+          <DisabledLabel />
+        </div>
+      )}
+    </div>
+  )
+}
+
+type SalesChannelsHeaderProps = {
+  openCreateModal: () => void
+  filterText?: string
+  setFilterText: (text: string) => void
+}
+
+/**
+ * Sales channel header.
+ */
+function SalesChannelsHeader(props: SalesChannelsHeaderProps) {
+  const { openCreateModal, filterText, setFilterText } = props
+  const [showFilter, setShowFilter] = useState(false)
+
+  const inputRef = useRef()
+
+  const classes = {
+    "translate-y-[-50px]": showFilter,
+    "translate-y-[0px]": !showFilter,
+  }
+
+  const hideFilter = () => {
+    setShowFilter(false)
+    setFilterText("")
+  }
+
+  useEffect(() => {
+    if (showFilter) {
+      // inputRef.current.focus()
+    }
+  }, [showFilter])
+
+  return (
+    <div className="h-[55px] mb-6 overflow-hidden">
+      <div className={clsx(" transition-all duration-200", classes)}>
+        <div className="h-[55px]">
+          <div className="flex justify-between items-center mb-1">
+            <h2 className="font-semibold text-xlarge text-grey-90">
+              Sales Channels
+            </h2>
+            <div className="flex justify-between items-center gap-4">
+              <SearchIcon
+                size={15}
+                onClick={() => setShowFilter(true)}
+                className="cursor-pointer"
+              />
+              <PlusIcon
+                size={15}
+                onClick={openCreateModal}
+                className="cursor-pointer"
+              />
+            </div>
+          </div>
+          <div className="text-grey-50 text-small mb-6">
+            Control which products are available in which channels
+          </div>
+        </div>
+
+        <div className="h-[40px] my-[5px] w-full flex items-center justify-around gap-2 text-grey-40 bg-grey-5 px-4 rounded-xl border">
+          <SearchIcon size={20} />
+          <input
+            ref={inputRef}
+            value={filterText}
+            onChange={(e) => setFilterText(e.target.value)}
+            placeholder="Search by title or description"
+            className="bg-inherit outline-none outline-0 w-full remove-number-spinner leading-base text-grey-90 font-normal caret-violet-60 placeholder-grey-40"
+            autoComplete="off"
+          />
+          <CrossIcon onClick={hideFilter} className="cursor-pointer" />
+        </div>
+      </div>
+    </div>
+  )
+}
+
+type SalesChannelsListProps = {
+  activeChannelId: string
+  openCreateModal: () => void
+  filterText?: string
+  setFilterText: (text: string) => void
+  salesChannels: SalesChannel[]
+  setActiveSalesChannel: (sc: SalesChannel) => void
+}
+
+/**
+ * Sales channels list.
+ */
+function SalesChannelsList(props: SalesChannelsListProps) {
```

Comment:
used it initially but there were some issues with the indicator

Reply:
What was the issue?

Reply:
I tried using RadioGroupPrimitive.Indicator and there were some issues with styling(of active state), but you are probably referring to our `RadioGroup` component which I didn't use because I considered this component to be more of a list/list item type